source/tools/BitSet.h

```diff
@@ -610,11 +618,28 @@ namespace emp {
 
 }
 
-template <size_t NUM_BITS> std::ostream & operator<<(std::ostream & out, const emp::BitSet<NUM_BITS> & _bit_set) {
-  _bit_set.Print(out);
-  return out;
+/// For hashing BitSets
+namespace std
```

Comment:
Technically, this is the "correct" way to implement various features (including hash) that are supposed to automatically interact with std::, and your intuition is correct for why it's allowed -- it only works when interacting with the namespace you are defining.  (Usually, that is; you can have some strange interactions crop up due to implicit conversions).

Fundamentally, this was a poor design decision on the part of std::unordered_set, and I do expect it to be fixed in one of the next couple of C++ updates, but alas not yet.  (Mostly because there's not an obvious fix; all of the other solutions do technically have drawbacks even if they are clearly better than the current solution -- they just want to limit it to one more fix and be done with it).